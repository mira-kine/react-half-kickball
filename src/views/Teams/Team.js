import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import { useParams } from 'react-router-dom/cjs/react-router-dom.min';
import Loading from '../../components/loading/Loading';
import { getTeamById } from '../../services/teams';

function Team() {
  const [team, setTeam] = useState({});
  const [loading, setLoading] = useState(true);
  const params = useParams();

  useEffect(() => {
    const fetchData = async () => {
      const resp = await getTeamById(params.id);
      setTeam(resp);
      setLoading(false);
    };
    fetchData();
  }, [params.id]);

  return loading ? (
    <Loading />
  ) : (
    <>
      <Link to="/teams">Teams</Link> &raquo; <span>{team.name}</span>
      <h1>{team.name}</h1>
      <p>
        {team.city}, {team.state}
      </p>
      <p>
        <Link to={`/teams/${params.id}/edit`}>Edit Team</Link>
      </p>
    </>
  );
}

export default Team;
