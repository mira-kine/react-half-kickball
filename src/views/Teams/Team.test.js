import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';
import Team from './Team';

it('should render a teamâ€™s detail page', async () => {
  // MemoryRouter is needed because <Team> uses the <Link> component
  const { container } = render(
    <MemoryRouter initialEntries={['/teams/2/edit']}>
      <Route path="/teams/:id/edit" component={Team} />
    </MemoryRouter>
  );
  screen.getByLabelText(/loading/);
  await screen.findAllByText(/Bridge City Sneakers/);

  expect(container).toMatchSnapshot();
});
