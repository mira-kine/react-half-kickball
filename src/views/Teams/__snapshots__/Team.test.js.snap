// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a team’s detail page 1`] = `
<div>
  <a
    href="/teams"
  >
    Teams
  </a>
   » 
  <span>
    Bridge City Sneakers
  </span>
  <h1>
    Bridge City Sneakers
  </h1>
  <p>
    Portland
    , 
    OR
  </p>
  <p>
    <a
      href="/teams/2/edit"
    >
      Edit Team
    </a>
  </p>
</div>
`;
